src/super_gradients/training/models/segmentation_models/ppliteseg.py

```diff
@@ -0,0 +1,383 @@
+                 decoder_up_factors: List[int],
+                 decoder_channels: List[int],
+                 decoder_upsample_mode: Union[UpsampleMode, str],
+                 head_upsample_mode: Union[UpsampleMode, str],
+                 head_mid_channels: int,
+                 dropout: float,
+                 use_aux_heads: bool,
+                 aux_hidden_channels: List[int],
+                 aux_scale_factors: List[int]
+                 ):
+        """
+        :param backbone: Backbone nn.Module should implement the abstract class `AbstractSTDCBackbone`.
+        :param projection_channels_list: channels list to project encoder features before fusing with the decoder
+            stream.
+        :param sppm_inter_channels: num channels in each sppm pooling branch.
+        :param sppm_out_channels: The number of output channels after sppm module.
+        :param sppm_pool_sizes: spatial output sizes of the pooled feature maps.
+        :param sppm_upsample_mode: Upsample mode to original size after pooling.
+        :param :
+        :param :
+        """
+        super().__init__(use_aux_heads=use_aux_heads)
+
+        # Init Encoder
+        backbone_out_channels = backbone.get_backbone_output_number_of_channels()
+        assert len(backbone_out_channels) == len(projection_channels_list), \
+            f"The length of backbone outputs ({backbone_out_channels}) should match the length of projection channels" \
+            f"({len(projection_channels_list)})."
+        context = SPPM(in_channels=backbone_out_channels[-1],
+                       inter_channels=sppm_inter_channels,
+                       out_channels=sppm_out_channels,
+                       pool_sizes=sppm_pool_sizes,
+                       upsample_mode=sppm_upsample_mode,
+                       align_corners=align_corners)
+        self.encoder = PPLiteSegEncoder(backbone=backbone,
+                                        context_module=context,
+                                        projection_channels_list=projection_channels_list)
+        encoder_channels = self.encoder.get_output_number_of_channels()
+
+        # Init Decoder
+        self.decoder = PPLiteSegDecoder(encoder_channels=encoder_channels,
+                                        up_factors=decoder_up_factors,
+                                        out_channels=decoder_channels,
+                                        upsample_mode=decoder_upsample_mode,
+                                        align_corners=align_corners)
+
+        # Init Segmentation classification heads
+        self.seg_head = nn.Sequential(
+            SegmentationHead(in_channels=decoder_channels[-1],
+                             mid_channels=head_mid_channels,
+                             num_classes=num_classes,
+                             dropout=dropout),
+            make_upsample_module(scale_factor=8, upsample_mode=head_upsample_mode, align_corners=align_corners)
```

Comment:
is it always 8? even if we define a different backbone?

Reply:
Added a scale_factor arg in case a custom design is required. [2972d7a](https://github.com/Deci-AI/super-gradients/pull/349/commits/2972d7aa52b156c2fafca28a4437ab7b63c8cc8b)